conf/configuration.go

```diff
@@ -87,7 +88,8 @@ type Configuration struct {
 		Templates   EmailContentConfiguration `json:"templates"`
 		URLPaths    EmailContentConfiguration `json:"url_paths"`
 	} `json:"mailer"`
-	External ExternalProviderConfiguration `json:"external"`
+	External       ExternalProviderConfiguration `json:"external"`
+	DisableSignups bool                          `json:"disable_signups"`
```

Comment:
Normally I would agree, but in this case I think the UX for the end user is clearer by specifying that you want to disable signup. It also plays a little better with env variables imo...

Reply:
> I think the UX for the end user is clearer by specifying that you want to disable signup

IMHO `DISABLE_SIGNUPS=true` and `SIGNUP_ENABLED=false` are equally clear. The latter would be consistent with the rest of the configuration too. 

> It also plays a little better with env variables imo

Can you be specific about how it plays better? We do have support for defaults from env vars if that is what you referencing.

Reply:
Just tested it, and out of the box the default true doesn't work with a boolean.

used:

```
SignupEnabled bool                          `json:"signup_enabled" default:"true"`
```

But when I run:

```
GOTRUE_SIGNUP_ENABLED=false go run main.go 
```

The settings page still gives me:

```
{"external":{"bitbucket":false,"github":false,"gitlab":false,"google":false},"signup_enabled":true,"autoconfirm":false}
```

This is because the defaults are not set based on whether the ENV var exists or not, but based on whether the value is empty - and `false` is the empty value for a bool in go.

that's what I mean with `DISABLE_SIGNUPS=true` playing better with env variables

Reply:
Actually, that is because the env var is `GOTRUE_SIGNUPENABLED`. If you want the extra underscore, you need `split_words:"true"` added to the tags. 

envconfig specifically looks for set values, and doesn't compare to the zero value. https://github.com/kelseyhightower/envconfig/blob/master/envconfig.go#L151-L163

Reply:
Makes a lot of sense - PR coming with this change

Reply:
Thanks for being awesome 🙌